pkg/operator/serviceaccountissuercontroller/serviceaccountissuer_controller.go

```diff
@@ -60,31 +73,27 @@ func (c *ServiceAccountIssuerController) sync(ctx context.Context, controllerCon
 
 	// there is no service account issuer set and there are no service account issuers in status, no-op.
 	if len(currentIssuer) == 0 && len(operator.Status.ServiceAccountIssuers) == 0 {
-		return nil
-	}
-
-	// if the status is not set, but the current issuer is not empty, update the status and let the controller resync
-	if len(currentIssuer) > 0 && len(operator.Status.ServiceAccountIssuers) == 0 {
 		operatorCopy := operator.DeepCopy()
-		operatorCopy.Status.ServiceAccountIssuers = []operatorv1.ServiceAccountIssuerStatus{
-			{
-				Name: currentIssuer,
-			},
-		}
+		operatorCopy.Status.ServiceAccountIssuers = defaultServiceAccountIssuerValue
 		_, statusUpdateErr := c.kubeAPIServerOperatorClient.UpdateStatus(ctx, operatorCopy, metav1.UpdateOptions{})
 		if statusUpdateErr == nil {
-			klog.Infof("ServiceAccountIssuer %q is now set as active issuer", currentIssuer)
+			klog.Infof("ServiceAccountIssuer %q is set as a default value", defaultServiceAccountIssuerValue[0].Name)
 			statusUpdateErr = factory.SyntheticRequeueError
 		}
 		return statusUpdateErr
 	}
 
-	// in this case, user deleted service account issuer value, which means we have to remove all values from the status
-	// if we don't do this and instead just remove first value, the new tokens will be issued with "previous" service
-	// account issuer which is not desired
-	if len(currentIssuer) == 0 && len(operator.Status.ServiceAccountIssuers) != 0 {
+	// this is a case when issuer is not set in auth config and the operator status already has the default issuer set.
+	if isDefaultServiceAccountIssuer(currentIssuer, operator.Status.ServiceAccountIssuers) {
+		return nil
+	}
+
+	// in this case the issuer was previously set in auth config, and later it was removed.
+	// this case must result into having default issuer set.
+	// this allows to instantly remove compromised issuers if needed.
+	if len(currentIssuer) == 0 {
```

Comment:
the below code only differs from the `if len(currentIssuer) == 0 && len(operator.Status.ServiceAccountIssuers) == 0 {` case by a different logging message.

I wonder if grouping them into a single branch could make this easier to read.

Reply:
`len(operator.Status.ServiceAccountIssuers) == 0` should never happen as that means there is no issuer set :-)

Reply:
`len(operator.Status.ServiceAccountIssuers) == 0` is the initial state

Reply:
gotcha, I combined those two cases, unit tests seems to pass :-)